src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_symmetric.cpp

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_symmetric.h"
+
+static_assert(PAL_OperationEncrypt == kCCEncrypt, "");
+static_assert(PAL_OperationDecrypt == kCCDecrypt, "");
+
+static_assert(PAL_AlgorithmAES == kCCAlgorithmAES128, "");
+static_assert(PAL_Algorithm3DES == kCCAlgorithm3DES, "");
+
+static_assert(PAL_ChainingModeECB == kCCModeECB, "");
+static_assert(PAL_ChainingModeCBC == kCCModeCBC, "");
+
+static_assert(PAL_PaddingModeNone == ccNoPadding, "");
+static_assert(PAL_PaddingModePkcs7 == ccPKCS7Padding, "");
+
+// No PAL_SymmetricOptions are currently mapped, so no asserts required.
+
+extern "C" void AppleCryptoNative_CryptorFree(CCCryptorRef cryptor)
+{
+    if (cryptor != nullptr)
+    {
+        CCCryptorRelease(cryptor);
+    }
+}
+
+extern "C" int AppleCryptoNative_CryptorCreate(PAL_SymmetricOperation operation,
+                                               PAL_SymmetricAlgorithm algorithm,
+                                               PAL_ChainingMode chainingMode,
+                                               PAL_PaddingMode paddingMode,
+                                               const uint8_t* pbKey,
+                                               int32_t cbKey,
+                                               const uint8_t* pbIv,
+                                               PAL_SymmetricOptions options,
+                                               CCCryptorRef* ppCryptorOut,
+                                               int32_t* pkCCStatus)
+{
+    if (pbKey == nullptr || cbKey < 1 || ppCryptorOut == nullptr)
+        return -1;
```

Comment:
The C# call sites are using out for things like errorCode (pkCCStatus).  We should always set it to a value here (and in other such cases), or else it could end up containing garbage.